### PR TITLE
dashboard/feature/add_logic_in_taxi_filter

### PR DESCRIPTION
**Pull request by Asiasama710**
-   I added fake data to the taxi in FakeRemoteGateway, so I can make a filter using them
-  to be able to do a filter on the data, I needed to add mapper, usecase and make request inside the screen module so
-  I added two interaction listeners, the first one is `save`, through which I perform the request
The other one, `cancel`, cancels the filter
-  and provide usecase in `di`

 **_Note_** : i have issues in paging with filtering so i resolve it in new PR 

https://github.com/TheChance101/beep-beep/assets/105972942/8c2d2b48-17ee-4685-9266-59784e975053

### CODE REVIEW

**Review by devsadeq — APPROVED** [review]
LGTM

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/data/remote/mapper/Taxi.kt
Comment (Saif-Alhaider) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/data/remote/mapper/Taxi.kt:1]: don't add all these mappers in this file add each mapper to the data class file
also the name indicates to the taxi entity in the domain

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/data/remote/mapper/Taxi.kt [hunk 1]
```diff
@@ -52,4 +67,10 @@ fun getTaxiStatus(status: Int) =
         0 -> TaxiStatus.OFFLINE
         1 -> TaxiStatus.ONLINE
         else -> TaxiStatus.ON_RIDE
+    }
+fun setTaxiStatus(status: TaxiStatus) =
+    when (status) {
+        TaxiStatus.OFFLINE -> 0
+        TaxiStatus.ONLINE -> 1
+        else -> 2
```
Comment (Saif-Alhaider) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/data/remote/mapper/Taxi.kt:75]: add it inside a class is better

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/taxi/TaxiScreenModel.kt [hunk 2]
```diff
@@ -144,12 +146,13 @@ class TaxiScreenModel(
...
         updateState { it.copy(isAddNewTaxiDialogVisible = true) }
     }
+
     //endregion
 
     //region filter menu listener
 
     override fun onFilterMenuDismiss() {
-        updateState { it.copy(isFilterDropdownMenuExpanded = false) }
+        updateState { it.copy(isFilterDropdownMenuExpanded = false) }//no need
```
Comment (Saif-Alhaider) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/presentation/taxi/TaxiScreenModel.kt:155]: no need for what ?
give an explanation please

**Review by Saif-Alhaider — COMMENTED** [review]
i'm requesting a few changes you've done well 🔥🌷

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/domain/entity/TaxiFiltered.kt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package org.thechance.common.domain.entity
+
+import org.thechance.common.domain.util.TaxiStatus
+
+data class TaxiFiltered(
```
Comment (mujtaba912) [inline client_dashboard/common/src/commonMain/kotlin/org/thechance/common/domain/entity/TaxiFiltered.kt:5]: TaxiFiltered make it seem like the taxi is already filtered but this is the data class that give the parameters that filter the taxi so a better name for it is TaxiFilter

**Review by mujtaba912 — CHANGES_REQUESTED** [review]

File: client_dashboard/common/src/commonMain/kotlin/org/thechance/common/domain/entity/TaxiFiltered.kt (hunk 3 above)
Comment (Asiasama710) [reply client_dashboard/common/src/commonMain/kotlin/org/thechance/common/domain/entity/TaxiFiltered.kt:5]: but filter is verb

**Review by mujtaba912 — APPROVED** [review]

**Pull request merged**